enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,135 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@egarcia"
+  - "@adduarte"
+reviewers:
+  - "@mandre"
+  - "@pierre"
+approvers:
+  - "@mandre"
+  - "@pierre"
+creation-date: 2020-5-26
+last-updated: 2020-5-26
+status: implementable
+---
+
+# OpenStack BYO External Connectivity
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up external connectivity to the OpenShift cluster. Depending on how the OpenStack cluster
+is set up, the use of floating IPs may be impossible.
+
+## Motivation
+
+### Goals
+
+- Extend the installer to support provider networks
```

Comment:
IIUC we won't be able to fully support provider networks until we make the switch to the external cloud provider. That's because provider network likely does not have nova-metadata agent and it's currently a requirement for the in-tree openstack cloud provider.
After switching to the external cloud provider, we'll be able to drop the nova-metadata requirement and use config-drive instead.

Reply:
I asked same question because the in tree hard code to metadata and don't support config drive

so the questions come to when the external cloud provider will be considered?

Reply:
We'd like to switch to external cloud provider in openshift 4.7. From what I understand, the in-tree cloud provider will be removed from the k8s code base in the 1.21 cycle, mapping to openshift 4.8 so we'll have to make the switch eventually. And we already have our hands full for 4.6.